src/Runner.Worker/ContainerOperationProvider.cs

```diff
@@ -238,35 +195,7 @@ private async Task StartContainerAsync(IExecutionContext executionContext, Conta
 
             if (container.IsJobContainer)
             {
-                // Configure job container - Mount workspace and tools, set up environment, and start long running process
-                var githubContext = executionContext.ExpressionValues["github"] as GitHubContext;
-                ArgUtil.NotNull(githubContext, nameof(githubContext));
-                var workingDirectory = githubContext["workspace"] as StringContextData;
-                ArgUtil.NotNullOrEmpty(workingDirectory, nameof(workingDirectory));
-                container.MountVolumes.Add(new MountVolume(HostContext.GetDirectory(WellKnownDirectory.Work), container.TranslateToContainerPath(HostContext.GetDirectory(WellKnownDirectory.Work))));
-#if OS_WINDOWS
-                container.MountVolumes.Add(new MountVolume(HostContext.GetDirectory(WellKnownDirectory.Externals), container.TranslateToContainerPath(HostContext.GetDirectory(WellKnownDirectory.Externals))));
-#else
-                container.MountVolumes.Add(new MountVolume(HostContext.GetDirectory(WellKnownDirectory.Externals), container.TranslateToContainerPath(HostContext.GetDirectory(WellKnownDirectory.Externals)), true));
-#endif
-                container.MountVolumes.Add(new MountVolume(HostContext.GetDirectory(WellKnownDirectory.Temp), container.TranslateToContainerPath(HostContext.GetDirectory(WellKnownDirectory.Temp))));
-                container.MountVolumes.Add(new MountVolume(HostContext.GetDirectory(WellKnownDirectory.Actions), container.TranslateToContainerPath(HostContext.GetDirectory(WellKnownDirectory.Actions))));
-                container.MountVolumes.Add(new MountVolume(HostContext.GetDirectory(WellKnownDirectory.Tools), container.TranslateToContainerPath(HostContext.GetDirectory(WellKnownDirectory.Tools))));
-
-                var tempHomeDirectory = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Temp), "_github_home");
-                Directory.CreateDirectory(tempHomeDirectory);
-                container.MountVolumes.Add(new MountVolume(tempHomeDirectory, "/github/home"));
-                container.AddPathTranslateMapping(tempHomeDirectory, "/github/home");
-                container.ContainerEnvironmentVariables["HOME"] = container.TranslateToContainerPath(tempHomeDirectory);
-
-                var tempWorkflowDirectory = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Temp), "_github_workflow");
-                Directory.CreateDirectory(tempWorkflowDirectory);
-                container.MountVolumes.Add(new MountVolume(tempWorkflowDirectory, "/github/workflow"));
-                container.AddPathTranslateMapping(tempWorkflowDirectory, "/github/workflow");
-
-                container.ContainerWorkDirectory = container.TranslateToContainerPath(workingDirectory);
-                container.ContainerEntryPoint = "tail";
-                container.ContainerEntryPointArgs = "\"-f\" \"/dev/null\"";
+                MountWellKnownDirectories(executionContext, container);
```

Comment:
same comment about whether this is just copy-paste? undo if so, to reduce the diff in code review.